src/main/java/org/erp/school/client/services/impl/DefaultClientService.java

```diff
@@ -0,0 +1,58 @@
+package org.erp.school.client.services.impl;
+
+import org.erp.school.client.Client;
+import org.erp.school.client.dto.ClientDTO;
+import org.erp.school.client.exception.ClientNotFoundException;
+import org.erp.school.client.repository.ClientRepository;
+import org.erp.school.client.services.ClientService;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import java.net.URI;
+
+@Service
+public class DefaultClientService implements ClientService {
```

Comment:
What is the reason to call DefaultClientService? I would have ClientService and ClientServiceImpl.

Reply:
You can have potentially more than one implementation. ClientServiceImpl wouldn't really explain a lot. I am also not a big fan of adding Impl at the end since you would see that it implements service, but again I can change it if you would come up with a meaningful name.

Reply:
ok. I understood.
No problem.